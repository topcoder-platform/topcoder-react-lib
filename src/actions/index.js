import authActions from './auth';
import smpActions from './smp';
import statsActions from './stats';
import termsActions from './terms';
import directActions from './direct';
import groupActions from './groups';
import errorActions from './errors';
import challengeActions from './challenge';
import profileActions from './profile';
import memberActions from './members';
import memberTaskActions from './member-tasks';
import reviewOpportunityActions from './reviewOpportunity';
import lookupActions from './lookup';

export const actions = {
  auth: authActions.auth,
  smp: smpActions.smp,
  challenge: challengeActions.challenge,
  terms: termsActions.terms,
  stats: statsActions.stats,
  direct: directActions.direct,
  groups: groupActions.groups,
  errors: errorActions.errors,
  profile: profileActions.profile,
  members: memberActions.members,
  memberTasks: memberTaskActions.memberTasks,
  reviewOpportunity: reviewOpportunityActions.reviewOpportunity,
  lookup: lookupActions.lookup,
};

export default undefined;
