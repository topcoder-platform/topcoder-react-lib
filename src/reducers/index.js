/**
 * Export reducers.
 */
import { redux } from 'topcoder-react-utils';

import auth, { factory as authFactory } from './auth';
import stats, { factory as statsFactory } from './stats';
import terms, { factory as termsFactory } from './terms';
import direct, { factory as directFactory } from './direct';
import groups, { factory as groupsFactory } from './groups';
import errors, { factory as errorsFactory } from './errors';
import challenge, { factory as challengeFactory } from './challenge';
import profile, { factory as profileFactory } from './profile';
import members, { factory as membersFactory } from './members';
import lookup, { factory as lookupFactory } from './lookup';
import memberTasks, { factory as memberTasksFactory } from './member-tasks';
import reviewOpportunity, { factory as reviewOpportunityFactory }
  from './reviewOpportunity';
import mySubmissionsManagement, { factory as mySubmissionsManagementFactory }
  from './my-submissions-management';


export function factory(options) {
  return redux.resolveReducers({
    auth: authFactory(options),
    stats: statsFactory(options),
    terms: termsFactory(options),
    direct: directFactory(options),
    groups: groupsFactory(options),
    errors: errorsFactory(options),
    challenge: challengeFactory(options),
    profile: profileFactory(options),
    lookup: lookupFactory(options),
    members: membersFactory(options),
    memberTasks: memberTasksFactory(options),
    reviewOpportunity: reviewOpportunityFactory(options),
    mySubmissionsManagement: mySubmissionsManagementFactory(options),
  });
}

export default ({
  auth,
  stats,
  terms,
  direct,
  groups,
  errors,
  challenge,
  profile,
  lookup,
  members,
  memberTasks,
  reviewOpportunity,
  mySubmissionsManagement,
});
