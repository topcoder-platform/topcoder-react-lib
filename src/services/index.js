/**
 * Export services.
 */
import * as api from './api';
import * as billing from './billing';
import * as challenge from './challenges';
import * as direct from './direct';
import * as groups from './groups';
import * as members from './members';
import * as terms from './terms';
import * as communities from './communities';
import * as reviewOpportunities from './reviewOpportunities';
import * as userSetting from './user-settings';
import * as user from './user';
import * as lookup from './lookup';

export const services = {
  api,
  billing,
  terms,
  communities,
  challenge,
  direct,
  groups,
  members,
  user,
  userSetting,
  reviewOpportunities,
  lookup,
};

export default undefined;
