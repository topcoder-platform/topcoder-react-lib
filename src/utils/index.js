/**
 * Export utils.
 */
import logger from './logger';
import * as tc from './tc';
import * as time from './time';
import * as mock from './mock';
import * as errors from './errors';
import * as filter from './challenge/filter';
import * as buckets from './challenge/buckets';
import * as sort from './challenge/sort';
import * as url from './url';
import * as submission from './submission';

const challenge = {
  filter,
  buckets,
  sort,
};

export {
  challenge,
  logger,
  tc,
  time,
  mock,
  errors,
  url,
  submission,
};
