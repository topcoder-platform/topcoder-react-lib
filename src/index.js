/**
 * Export the lib.
 */
import reducers, { factories as reducerFactories, factory as reducerFactory } from './reducers';

export { reducerFactories, reducerFactory };

export { reducers };

export { actions } from './actions';

export { services } from './services';

export { challenge, logger, errors, tc, time, mock } from './utils';
