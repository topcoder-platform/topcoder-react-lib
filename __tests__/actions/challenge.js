import { redux } from 'topcoder-react-utils';
import { actions } from '../../src';

jest.mock('../../src/services/challenges');

const mockFetch = resolvesTo => jest.fn(() =>
  Promise.resolve({ json: () => resolvesTo }));

let originalFetch;

beforeAll(() => {
  originalFetch = global.fetch;
});

afterAll(() => {
  global.fetch = originalFetch;
});

describe('challenge.fetchChallengeInit', () => {
  const a = actions.challenge.getDetailsInit(12345);

  test('has expected type', () => {
    expect(a.type).toBe('CHALLENGE/GET_DETAILS_INIT');
  });

  test('payload is the challenge ID, converted to string, if necessary', () =>
    expect(a.payload).toBe('12345'));
});

describe('challenge.fetchSubmissionsInit', () => {
  const a = actions.challenge.getSubmissionsInit(12345);

  test('has expected type', () => {
    expect(a.type).toBe('CHALLENGE/GET_SUBMISSIONS_INIT');
  });

  test('payload is challengeId', () =>
    expect(a.payload).toBe('12345'));
});

describe('challenge.getDetailsDone', () => {
  global.fetch = mockFetch({ result: { content: ['DUMMY DATA'] } });

  const a = actions.challenge.getDetailsDone(12345);

  test('has expected type', () => {
    expect(a.type).toBe('CHALLENGE/GET_DETAILS_DONE');
  });

  const mockChallenge =
    require('../../src/services/__mocks__/data/challenge-normalized.json');

  test('payload is a promise which resolves to the expected object', () =>
    a.payload.then((res) => {
      res.communities = Array.from(res.communities);
      expect(res).toEqual(mockChallenge);
    }));
});

describe('challenge.fetchSubmissionsDone', () => {
  global.fetch = mockFetch({
    challengeId: '12345',
    submissions: 'DUMMY DATA',
  });

  const a = actions.challenge.getSubmissionsDone(12345, {});

  test('has expected type', () => {
    expect(a.type).toBe('CHALLENGE/GET_SUBMISSIONS_DONE');
  });

  test('payload is a promise which resolves to the expected object', () =>
    a.payload.then(res => expect(res).toEqual({
      challengeId: '12345',
      submissions: 'DUMMY DATA',
    })));
});

test('challenge.getActiveChallengesCountInit', () => {
  const actionResult = actions.challenge.getActiveChallengesCountInit();
  expect(actionResult).toMatchSnapshot();
});

test('challenge.getActiveChallengesCountDone', async () => {
  const actionResult =
    await redux.resolveAction(actions.challenge.getActiveChallengesCountDone('handle', 'tokenV3'));
  expect(actionResult).toMatchSnapshot();
});
