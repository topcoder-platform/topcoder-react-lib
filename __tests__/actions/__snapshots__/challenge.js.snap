// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`challenge.getActiveChallengesCountDone 1`] = `
Object {
  "payload": 10,
  "type": "CHALLENGE/GET_ACTIVE_CHALLENGES_COUNT_DONE",
}
`;

exports[`challenge.getActiveChallengesCountInit 1`] = `
Object {
  "type": "CHALLENGE/GET_ACTIVE_CHALLENGES_COUNT_INIT",
}
`;
